core/domain/user_services.py

```diff
@@ -435,11 +435,11 @@ def generate_initial_profile_picture(user_id):
         user_id: str. The unique ID of the user.
     """
     user_email = get_email_from_user_id(user_id)
-    user_gravatar = fetch_gravatar(user_email)
+    user_gravatar = fetch_gravatar_by_email(user_email)
```

Comment:
There should be some alternative function if you do so for users without emails; what is the plan for gravatars for that case?

Reply:
Android app wouldn't use gravatars for profile picture generation, rather an in app method. So it doesn't matter for them what we use for gravatar generation here. See [this comment](https://docs.google.com/document/d/1sQagy3irkjB1Kt5sdb_KkOgxODt2m2ya0ZCoSJNlY4A/edit?disco=AAAAGvc_2B0) from design doc

For web we currently only have google id sign in, which needs email. And presently we don't know what attributes other auth methods might want to store, so I thought renaming this function would be the best option to tell the future developer about the functioning of these gravatar generating functions. I could go ahead and change these to work by user_id, but I thought it's best to leave it to future decision on that. Renaming also doesn't change anything for current system.

Reply:
I think conceptually you should still keep it as-is and have it handle the Android case properly later on.

Do the naming changes only when you actually implement the changes that require the renaming.

Reply:
Thanks for your suggestions. Will keep them in mind in future.
True that we won't be actually using gravatar now, so reverted those name changes.